src/List/Selection.elm

```diff
@@ -115,3 +151,43 @@ map fn (Selection selected items) =
     Selection
         (Maybe.map fn selected)
         (List.map fn items)
+
+
+{-| Apply a function to only the selected item, if any item is selected.
+-}
+mapSelected : (a -> a) -> Selection a -> Selection a
+mapSelected fn (Selection selected items) =
+    Selection
+        (Maybe.map fn selected)
+        items
+
+
+{-| Decode a selection from JSON. The result won't have any item
+selected.
+
+If you need to decode the selection too, use `map2`. Here's an
+example, assuming that you have "selected" and "items" in your JSON:
+
+    import Json.Decode exposing (Decoder, decodeString, field, int, map2)
+
+
+    yourDecoder : Decoder (Selection Int)
+    yourDecoder =
+        map2 select
+            (field "selected" int)
+            (field "items" (decoder int))
+
+
+    result : Result String (Selection Int)
+    result =
+        decodeString yourDecoder
+            """{"selected": 2, "items": [1, 2, 3]}"""
+
+    Result.map selected result --> Ok (Just 2)
+
+    Result.map toList result --> Ok [1, 2, 3]
+
+-}
+decoder : Decoder a -> Decoder (Selection a)
+decoder =
+    Decode.list >> Decode.map fromList
```

Comment:
IMHO this should be in an extra module. `fromList` is exposed and this function is trivial (no need to access the opaque type or anything).